### PR TITLE
chore: add underscore to proposedName when it is equal with type or be literal

### PR DESCRIPTION
**Pull request by whalecold**
**What this PR does / why we need it**:

chore: add underscore to proposedName when it is equal with type or be literal

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by whalecold** [comment]
@iawia002

File: utils/generators/golang/helper.go [hunk 1]
```diff
@@ -433,6 +433,12 @@ func (h *helper) Functions() ([]function, []string) {
 					ProposedName: sigNames.proposeName("", result.Type),
 					Typ:          h.namer.Name(result.Type),
 				}
+
+				// name may be `int` `string` etc.
```
Comment (iawia002) [inline utils/generators/golang/helper.go:437]: 可以在 `sigNames.proposeName` 这个函数里面一起做，这个里面本身也有这种判断

Comment (whalecold) [reply utils/generators/golang/helper.go]: done.

**Pull request merged**